### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by Russel1lTan**
Hello Teaching Team,

Here is my pull request to merge my topic-recognition into Shakes' topic-recognition. The task I picked was Segment the HipMRI Study on Prostate Cancer (see Appendix for link) using the processed 2D slices (2D images) available here with the 2D UNet [1] with all labels having a minimum Dice similarity coefficient of 0.75 on the test set on the prostate label.I managed to get an accuracy of  in the test split. The files included in this PR are as followed:

README.md file

recognition/2D_UNet_Segmentation_HongdaTan_47315240/: folder contains all the following:

dataset.py: contains data loading class

modules.py: contains all neural network architecture

train.py: contains main ability to train/save model and test

predict.py; contains scripts to load previous checkpoint and test that model

Data/: folder storing data should be located here

Thank you,
HongdaTan 47315240

### CODE REVIEW

**Comment by yexincheng** [comment]
**This is an initial inspection, no action is required at this point**
1. **Solves problem**: No any figures or metrics of training logs or prediction results.
2. **Implementation**: Did not save the checkpoint. Hyperparameters were hard-coded. model, criterion and optimizer were defined twice ([here](https://github.com/Russel1lTan/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_Segmentation_HongdaTan_47315240/train.py#L19C1-L22C50) and [here](https://github.com/Russel1lTan/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_Segmentation_HongdaTan_47315240/train.py#L31-L34)); might misunderstand model initalization. No evaluation metrics implemented.
3. **Design**: Good
4. **Comment**: Not enough comments. More detailed comments are required
5. **Difficulty**: Easy

**Note**
- Pull request title should be more informative.
- All files should be under your folder - 2D_UNet_Segmentation_HongdaTan_47315240, including readme, reference and license. 
- pycache should be commited

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. No evidance of model training provided.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. </td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>No Progressive commits. All on 1 day.</td><td>-3</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Train logs/charts needs to be added.</td><td>-3</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>Feedback action required: Feedback marks possible +2 if the requested changes are made. Remove cache files for merge.</td><td>-2</td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-22</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by gayanku** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by Russel1lTan** [comment]
> Feedback marks possible +2 if the requested changes are made (see above).

Hi, Gayan. I have deleted the cache files and merged. Is that all I need to do?
Thanks.

**Comment by shakes76** [comment]
Feedback +2

**Comment by shakes76** [comment]
Cant merge because of conflicting changes to main repo files (README). Please update for merge, doesn't affect grade.
